File: src/main/java/com/skyspacefiftyfive/service/TravelClassService.java
package com.skyspacefiftyfive.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.skyspacefiftyfive.domain.TravelClass;
import com.skyspacefiftyfive.dto.TravelClassDTO;
import com.skyspacefiftyfive.dto.TravelClassSearchDTO;
import com.skyspacefiftyfive.dto.TravelClassPageDTO;
import com.skyspacefiftyfive.dto.TravelClassConvertCriteriaDTO;
import com.skyspacefiftyfive.service.GenericService;
import com.skyspacefiftyfive.dto.common.RequestDTO;
import com.skyspacefiftyfive.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface TravelClassService extends GenericService<TravelClass, Integer> {

	List<TravelClass> findAll();

	ResultDTO addTravelClass(TravelClassDTO travelClassDTO, RequestDTO requestDTO);

	ResultDTO updateTravelClass(TravelClassDTO travelClassDTO, RequestDTO requestDTO);

    Page<TravelClass> getAllTravelClasss(Pageable pageable);

    Page<TravelClass> getAllTravelClasss(Specification<TravelClass> spec, Pageable pageable);

	ResponseEntity<TravelClassPageDTO> getTravelClasss(TravelClassSearchDTO travelClassSearchDTO);
	
	List<TravelClassDTO> convertTravelClasssToTravelClassDTOs(List<TravelClass> travelClasss, TravelClassConvertCriteriaDTO convertCriteria);

	TravelClassDTO getTravelClassDTOById(Integer travelClassId);







}

File: skyspacefiftyfive_ui/src/services/TravelClassService.js
import http from "../http-common"; 

class TravelClassService {
  getAllTravelClasss(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/travelClass/travelClasss`, searchDTO);
  }

  get(travelClassId) {
    return this.getRequest(`/travelClass/${travelClassId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/travelClass?field=${matchData}`, null);
  }

  addTravelClass(data) {
    return http.post("/travelClass/addTravelClass", data);
  }

  update(data) {
  	return http.post("/travelClass/updateTravelClass", data);
  }
  
  uploadImage(data,travelClassId) {
  	return http.postForm("/travelClass/uploadImage/"+travelClassId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new TravelClassService();
